pkg/agent/multicast/mcast_discovery.go

```diff
@@ -68,21 +68,7 @@ type IGMPSnooper struct {
 }
 
 func (s *IGMPSnooper) HandlePacketIn(pktIn *ofctrl.PacketIn) error {
-	matchers := pktIn.GetMatches()
-	// Get custom reasons in this packet-in.
-	match := openflow.GetMatchFieldByRegID(matchers, openflow.CustomReasonField.GetRegID())
-	if match == nil {
-		return fmt.Errorf("error getting match from IGMP marks in CustomField")
-	}
-	customReasons, err := openflow.GetInfoInReg(match, openflow.CustomReasonField.GetRange().ToNXRange())
-	if err != nil {
-		klog.ErrorS(err, "Received error while unloading customReason from OVS reg")
-		return err
-	}
-	if customReasons&openflow.CustomReasonIGMP == openflow.CustomReasonIGMP {
-		return s.processPacketIn(pktIn)
-	}
-	return nil
+	return s.processPacketIn(pktIn)
```

Comment:
We could move code of `processPacketIn` here.

Reply:
Done.